### ISSUE TITLE
False positive: `nested-min-max` drops inner `key=`/`default=` and suggests a non-equivalent call

### ISSUE THREAD

**Issue by Synvoya** [PyCQA/pylint#11130]
### Bug description

`nested-min-max` (W3301) flags a nested `min`/`max` even when the **inner** call carries a keyword argument (`key=`, `default=`), and the suggested replacement silently **drops that keyword**, changing the result.

```python
def f(a, b, c, key):
    return max(a, max(b, c, key=key))
```

pylint suggests `max(a, b, c)`, which is not equivalent:

```pycon
>>> k = lambda z: -z
>>> max(1, max(5, 3, key=k))   # original
3
>>> max(1, 5, 3)               # pylint's suggestion
5
```

The same happens on the splat path (`max(3, max(LIST, key=key))` → `max(3, *LIST)`).

### Command used

```shell
pylint --disable=all --enable=nested-min-max a.py
```

### Pylint output

```
a.py:3:11: W3301: Do not use nested call of 'max'; it's possible to do 'max(a, b, c)' instead (nested-min-max)
```

### Expected behavior

No message when the **inner** call has keyword arguments, since the flattened form is not equivalent. The outer-keyword case `max(a, max(b, c), key=k)` should still be flagged (there the key applies to all arguments).

### Root cause

`NestedMinMaxChecker.get_redundant_calls` selects inner calls to flatten using only `node.args` / `len(arg.parent.args) > 1` and never inspects `arg.keywords`; the flattening splice keeps only `arg.args`, so any keyword on the inner call is dropped.

### Pylint version

```
pylint 4.x (main branch), astroid 4.x
```
